lib/commanded/event/handler.ex

```diff
@@ -626,12 +626,38 @@ defmodule Commanded.Event.Handler do
       @doc false
       def before_reset, do: :ok
 
-      defoverridable init: 1, after_start: 1, before_reset: 0
+      @doc false
+      def error(error, failed_event, failure_context) do
+        retry_strategy = Keyword.get(unquote(using_opts), :retry, :stop)
+        Handler.handle_error(retry_strategy, error, failed_event, failure_context)
+      end
+
+      defoverridable init: 1, after_start: 1, before_reset: 0, error: 3
+    end
+  end
+
+  def handle_error(:stop, {:error, reason}, _failed_event, _failure_context) do
+    {:stop, reason}
+  end
+
+  def handle_error(:backoff, {:error, reason}, _event, %FailureContext{context: context}) do
+    context = Map.update(context, :failures, 1, fn failures -> failures + 1 end)
+    max_failures = Map.get(context, :max_failures, 2)
```

Comment:
From the context, should we do `{:backoff, [max_failures: 2]` as well?

1. Check the Context
2. Check the compile-time value

Reply:
I've changed the implementation, but I think we'll put a few common ones in the box, and let you provide your own module, or override error/3 so you can do custom stuff. WDYT?